pkg/infinispan/cluster.go

```diff
@@ -114,6 +116,55 @@ func (c Cluster) GracefulShutdown(podName string) error {
 	return validateResponse(rsp, reason, err, "during graceful shutdown", http.StatusNoContent)
 }
 
+// ISPN-13141 Upload custom task to perform graceful shutdown that does not fail on cache errors
+// This task calls Cache#shutdown which disables rebalancing on the cache before stopping it
+func (c Cluster) GracefulShutdownTask(podName string) error {
+	scriptName := "___org.infinispan.operator.gracefulshutdown.js"
+	url := fmt.Sprintf("%s/tasks/%s", consts.ServerHTTPBasePath, scriptName)
+	headers := map[string]string{"Content-Type": "text/plain"}
+	task := `
+	/* mode=local,language=javascript */
+	print("Executing operator shutdown task");
+	var System = Java.type("java.lang.System");
+	var HashSet = Java.type("java.util.HashSet");
+	var InternalCacheRegistry = Java.type("org.infinispan.registry.InternalCacheRegistry");
+
+	var stdErr = System.err;
+	var icr = cacheManager.getGlobalComponentRegistry().getComponent(InternalCacheRegistry.class);
+
+	var cacheNames = cacheManager.getCacheNames();
+	shutdown(cacheNames);
+
+	var internalCaches = new HashSet(icr.getInternalCacheNames());
+	/* The ___script_cache is included in both getCacheNames() and getInternalCacheNames so prevent repeated shutdown calls */
+	internalCaches.removeAll(cacheNames);
+	shutdown(internalCaches);
+
+	function shutdown(cacheNames) {
+	   var it = cacheNames.iterator();
+	   while (it.hasNext()) {
+			 name = it.next();
+			 print("Shutting down cache " + name);
+			 try {
+				cacheManager.getCache(name).shutdown();
+			 } catch (err) {
+				stdErr.println("Encountered error trying to shutdown cache " + name + ": " + err);
+			 }
+	   }
+	}
+	`
+	// Remove all new lines to prevent a "100 continue" response
```

Comment:
Probably utilizing go-rice could help, but make sense if this solution will be forever with us

Reply:
[ISPN-13141](https://issues.redhat.com/browse/ISPN-13141) will be fixed on the server-side, so this shouldn't be around for too long. Either way, we have to remove the new lines as we're sending the script via a curl string.